front/components/assistant/conversation/ConversationViewer.tsx

```diff
@@ -384,68 +393,102 @@ const ConversationViewer = React.forwardRef<
           <Spinner variant="color" size="xs" />
         </div>
       )}
-
       {conversation &&
-        groupedMessages.map((group) => {
-          return group.map((message) => {
-            return (
-              <MessageItem
-                key={`message-${message.sId}`}
-                conversationId={conversation.sId}
-                hideReactions={hideReactions}
-                isInModal={isInModal}
-                message={message}
-                owner={owner}
-                reactions={reactions}
-                ref={
-                  message.sId === prevFirstMessageId
-                    ? prevFirstMessageRef
-                    : undefined
-                }
-                user={user}
-                isLastMessage={latestPage?.messages.at(-1)?.sId === message.sId}
-                latestMentions={latestMentions}
-              />
-            );
-          });
+        typedGroupedMessages.map((typedGroup, index) => {
+          const isLastGroup = index === typedGroupedMessages.length - 1;
+          return (
+            <MessageGroup
+              key={`typed-group-${index}`}
+              messages={typedGroup}
+              isLastMessage={isLastGroup}
+            >
+              {typedGroup &&
+                typedGroup.map((group) => {
+                  return group.map((message) => {
+                    return (
+                      <MessageItem
+                        key={`message-${message.sId}`}
+                        conversationId={conversationId}
+                        hideReactions={hideReactions}
+                        isInModal={isInModal}
+                        message={message}
+                        owner={owner}
+                        reactions={reactions}
+                        ref={
```

Comment:
This logic was used for pagination but I guess it should not be moved to the MessageGroup level.